### ISSUE TITLE
Lots of errors when running on multiple computers

### ISSUE THREAD

**Issue by raghavmittal101** [RiflerRick/StockholmHotline#7]
We had a single server and two clients(physical laptops) setup for testing.
When connected with both client things were not working as expected.
Steps to reproduce:
- spin up a server in a laptop
- create one or two clients on physical laptops other then the host server laptop.
- try chatting
- disconnect and connect again is counted as new user. So on disconnection, the number of active connection counter is not decremented.

Snippet is from one client on one laptop. And log files are from multiple clients on other laptop.

server_log.txt is server log file

![capture](https://cloud.githubusercontent.com/assets/9378481/25567682/f1508e30-2e10-11e7-990e-f13287ed6cad.PNG)

[log.txt](https://github.com/RiflerRick/AnnoymousChatServer/files/966982/log.txt)
[log1.txt](https://github.com/RiflerRick/AnnoymousChatServer/files/966983/log1.txt)
[log2.txt](https://github.com/RiflerRick/AnnoymousChatServer/files/966984/log2.txt)

[server_log.txt](https://github.com/RiflerRick/AnnoymousChatServer/files/966985/server_log.txt)

**Comment by electron0zero** [comment]
@RiflerRick I was involved in testing.
if you need any help in the reproduction of this issue,
feel free to ask.

**Comment by RiflerRick** [comment]
yeah!!! i know about this disconnection issue. I actually did not code the system for a smooth disconnection. First of all I had to get the system working in situations. The system **at times** was not even working for multiple computers(clients) in the same network. I will deal with these later...

however you'll are free to make changes...just send me the PRs...
